jedi/plugins/stdlib.py

```diff
@@ -503,13 +504,16 @@ def get_signatures(self):
                 keys.add(key)
         return [PartialSignature(s, arg_count, keys) for s in func.get_signatures()]
 
+    def py__get__(self, instance, class_value):
+        return ValueSet([PartialObject(self._actual_value, self._arguments, self._instance)])
```

Comment:
The issue here is that you use `self._instance` instead of instance. This is also why you had to add the random boolean.

I'm working on it now locally, so please don't push further changes.